packages/ses/src/error/note-log-args.js

```diff
@@ -151,20 +151,24 @@ export const makeLRUCacheMap = budget => {
    */
   const set = (key, value) => {
     if (budget >= 1) {
+      /** @type {DoublyLinkedCell<WeakMap<object, V> | undefined> | undefined} */
       let cell = touchCell(key);
-      if (cell !== undefined) {
-        cell.keyValue.set(key, value);
-      } else {
+      if (cell === undefined) {
         if (size >= budget) {
           const condemned = head.prev;
           spliceOut(condemned); // Drop least recently used
+          condemned.data = undefined;
           size -= 1;
         }
         size += 1;
-        cell = makeSelfCell(key, value);
+        cell = makeSelfCell(new WeakMap());
         keyToCell.set(key, cell);
         spliceAfter(head, cell); // start most recently used
       }
+      if (!cell.data) {
+        throw TypeError('Internal error; expected cell.data');
```

Comment:
Actually this could totally happen when a key is pushed out then later re-introduced. We should simply reset `keyToCell` like if the cell was not found.